### DISCUSSION TITLE
How to add new app into the SolidOS Databrowser Webapp

### DISCUSSION THREAD

**Discussion by gp125fi** [solid/solidos#164] [Q&A]
Hi, is there any document which can demonstrate step by step to show how to add new app into the SolidOS Databrowser Webapp? And the authentication mechanism of the SolidOS Databrowser Webapp could work on new app too. If there isn’t this kind of document, then how to get this goal? For example, just like using address book, chat, dokieli, issue tracker, etc., I would like to add a solid-filemanager which may make the SolidOS Databrowser Webapp easier to manage file.

**Comment by jeff-zucker** [comment]
These are interesting questions. Currently the panes system is what is used to pull in apps.  There is a panes registry which lists the things  like chat, dokeli, etc. and a mechanism for writing panes such that each has a common setup - a label, a render method, etc. which allows them to be integrated into SolidOS.  I suggest you look in the [Solid Panes Repo](https://github.com/solidos/solid-panes) and come to the [SolidOS Chat Room](https://gitter.im/solid/solidos) if you have questions.

I am also working on a different kind of plugin system for apps which may be available soon, but is not usable right now.

The SolidOS team is also working on adding file-management so I suggest you come to meetings to talk over how we can best support each other rather than duplicating effort.

**Reply by gp125fi** [reply]
Thank you for your reply. Maybe this discussions area is a better place to discuss some technical question because it’s easier to search and find related subject than chat room. The efficiency of discussion in the SolidOS Chat Room seems rapider than this discussions area. I’ll go to the SolidOS Chat Room if I need more instant response. 

I’m looking forward to The SolidOS team’s documentation about demonstrating how to add new app into the SolidOS Databrowser Webapp. I’ll attend the meeting if I’m free next time.

**Comment by jeff-zucker** [comment]
> the authentication mechanism of the SolidOS Databrowser Webapp could work on new app too

Here's a potential problem  : SolidOS operates with full access - it can read/write/control everything in the pod.  If another app uses SolidOS's login, it will inherit all of SolidOS permissions.  We don't necessarily want photoAppXYZ to have full read/write/control over the entire pod.
